pyproject.toml

```diff
@@ -1,10 +1,73 @@
+[build-system]
+build-backend = "hatchling.build"
+requires = ["hatchling"]
+
+[lint.isort]
+known-first-party = ["litestar", "tests", "examples"]
+
+[lint.mccabe]
+max-complexity = 12
+
+[lint.pep8-naming]
+classmethod-decorators = [
+  "classmethod",
+  "pydantic.root_validator",
+  "pydantic.validator",
+  "sqlalchemy.ext.declarative.declared_attr",
+  "sqlalchemy.orm.declared_attr.directive",
+  "sqlalchemy.orm.declared_attr"
+]
+
+[lint.per-file-ignores]
+"docs/**/*.*" = ["S", "B", "DTZ", "A", "TCH", "ERA", "D", "RET"]
+"docs/examples/**" = ["T201"]
+"docs/examples/application_hooks/before_send_hook.py" = ["UP006"]
+"docs/examples/contrib/sqlalchemy/plugins/**/*.*" = ["UP006"]
+"docs/examples/data_transfer_objects**/*.*" = ["UP006"]
+"litestar/_openapi/schema_generation/schema.py" = ["C901"]
+"litestar/exceptions/*.*" = ["N818"]
+"litestar/handlers/**/*.*" = ["N801"]
+"litestar/params.py" = ["N802"]
+"test_apps/**/*.*" = ["D", "TRY", "EM", "S", "PTH"]
+"tests/**/*.*" = [
+  "A",
+  "ARG",
+  "B",
+  "BLE",
+  "C901",
+  "D",
+  "DTZ",
+  "EM",
+  "FBT",
+  "G",
+  "N",
+  "PGH",
+  "PIE",
+  "PLR",
+  "PLW",
+  "PTH",
+  "RSE",
+  "S",
+  "S101",
+  "SIM",
+  "TCH",
+  "TRY"
+]
+"tests/unit/test_contrib/test_sqlalchemy/**/*.*" = ["UP006"]
+"tests/unit/test_openapi/test_typescript_converter/test_converter.py" = ["W293"]
+"tools/**/*.*" = ["D", "ARG", "EM", "TRY", "G", "FBT"]
+"tools/prepare_release.py" = ["S603", "S607"]
+
+[lint.pydocstyle]
+convention = "google"
+
 [project]
```

Comment:
Is there some way to get the `project` section to be the first section in the file? Not a huge deal, but I just felt it probably makes sense to keep it as the first section.

Reply:
idk about this, i absolutely hate how it sorts everything, but having it consistent is cool. Ill look into it if i can solve it removing `[tool.ruff.` from a bunch of keys..